modules/grpc/bigquery/bigquery-worker.cpp

```diff
@@ -58,16 +88,308 @@ DestinationWorker::deinit()
+      if (!type_cast_to_int64(value->str, &v, NULL))
+        {
+          type_cast_drop_helper(owner->template_options.on_error, value->str, "integer");
+          return false;
+        }
+      reflection->SetInt64(message, field.field_desc, v);
+      break;
+    }
+    case FieldDescriptor::CppType::CPPTYPE_UINT32:
+    {
+      int64_t v;
+      if (!type_cast_to_int64(value->str, &v, NULL))
+        {
+          type_cast_drop_helper(owner->template_options.on_error, value->str, "integer");
+          return false;
+        }
+      reflection->SetUInt32(message, field.field_desc, (uint32_t) v);
+      break;
+    }
+    case FieldDescriptor::CppType::CPPTYPE_UINT64:
+    {
+      int64_t v;
+      if (!type_cast_to_int64(value->str, &v, NULL))
+        {
+          type_cast_drop_helper(owner->template_options.on_error, value->str, "integer");
+          return false;
+        }
+      reflection->SetUInt64(message, field.field_desc, (uint64_t) v);
+      break;
+    }
+    case FieldDescriptor::CppType::CPPTYPE_DOUBLE:
+    {
+      double v;
+      if (!type_cast_to_double(value->str, &v, NULL))
+        {
+          type_cast_drop_helper(owner->template_options.on_error, value->str, "double");
+          return false;
+        }
+      reflection->SetDouble(message, field.field_desc, v);
+      break;
+    }
+    case FieldDescriptor::CppType::CPPTYPE_FLOAT:
+    {
+      double v;
+      if (!type_cast_to_double(value->str, &v, NULL))
+        {
+          type_cast_drop_helper(owner->template_options.on_error, value->str, "double");
+          return false;
+        }
+      reflection->SetFloat(message, field.field_desc, (float) v);
+      break;
+    }
+    case FieldDescriptor::CppType::CPPTYPE_BOOL:
+    {
+      gboolean v;
+      if (!type_cast_to_boolean(value->str, &v, NULL))
+        {
+          type_cast_drop_helper(owner->template_options.on_error, value->str, "boolean");
+          return false;
+        }
+      reflection->SetBool(message, field.field_desc, v);
+      break;
+    }
+    default:
+      return false;
+    }
+
+success:
+  scratch_buffers_reclaim_marked(m);
+  return true;
+}
+
 LogThreadedResult
 DestinationWorker::insert(LogMessage *msg)
 {
+  DestinationDriver *owner = this->get_owner();
+
+  google::cloud::bigquery::storage::v1::ProtoRows *rows = this->current_batch.mutable_proto_rows()->mutable_rows();
+
+  google::protobuf::Message *message = owner->schema_prototype->New();
+  const google::protobuf::Reflection *reflection = message->GetReflection();
+
+  bool msg_has_field = false;
+  for (const auto &field : owner->fields)
+    {
+      bool field_inserted = this->insert_field(reflection, field, msg, message);
+      msg_has_field |= field_inserted;
+
+      if (!field_inserted && (owner->template_options.on_error & ON_ERROR_DROP_MESSAGE))
+        goto drop;
+    }
+
+  if (!msg_has_field)
+    goto drop;
+
+  this->batch_size++;
+  rows->add_serialized_rows(message->SerializeAsString());
```

Comment:
We should either check if the serialization succeeded or let this be a partial serialization. I had an abort when I had a `required` field, which got skipped because of an empty template resolution.